pkg/utils/release/controller.go

```diff
@@ -424,3 +450,132 @@ func (s *SuiteController) CreateReleasePipelineRoleBindingForServiceAccount(name
 	}
 	return roleBinding, nil
 }
+
+func (s *SuiteController) WaitForReleasePipelineToBeFinished(managedNamespace, devNamespace string, releaseName string, component *appstudioApi.Component, maxRetries int) error {
+	attempts := 1
+	var pr *v1beta1.PipelineRun
+	var prList *v1beta1.PipelineRunList
+
+	for {
+		err := wait.PollImmediate(20*time.Second, 30*time.Minute, func() (done bool, err error) {
+
+			if len(releaseName) > 0 {
```

Comment:
what? You are checking if the string length is greater than 0? Why?

Reply:
In order to use this function in the release.go test , there we have the release name as input .
 see the call of this function in [release.go(                        Expect(fw.AsKubeAdmin.ReleaseController.WaitForReleasePipelineToBeFinished(managedNamespace, devNamespace, releaseName, nil, 3)).To(Succeed())) 
 
 and 
 a call from another test where we don't have the Release name as a given  in  [e2e-test-default-with-deployment.go ](https://github.com/kasemAlem/e2e-tests/blob/a99fa875610203caa49acee66bcd3b1e74644094/tests/release/e2e-test-default-with-deployment.go#L188)

Reply:
so why aren't you just comparing to the empty string or something? Checking if a string parameter was provided by checking its length feels super weird to me

Reply:
https://github.com/kasemAlem/e2e-tests/blob/a99fa875610203caa49acee66bcd3b1e74644094/tests/release/e2e-test-default-with-deployment.go#L188 doesn't really make sense to me either. Why are we checking for a release PipelineRun in the managed namespace before checking if the Release exists? A release plr will never be created before a release pipelinerun... the creation of a release plr is part of reconciling the release cr. And if we got the release first, we would have its name